packages/react-components/lib/doors/door-marker.tsx

```diff
@@ -0,0 +1,222 @@
+  close: {
+    stroke: '#BC4812',
+  },
+  moving: {
+    stroke: '#E9CE9F',
+    strokeDasharray: 0.3,
+  },
+  unknown: {
+    stroke: 'grey',
+  },
+  transparent: {
+    stroke: 'transparent',
+  },
+});
+
+function useDoorStyle(doorState?: RomiCore.DoorState): string {
+  const classes = useDoorStyles();
+
+  if (!doorState) {
+    return classes.unknown;
+  }
+
+  switch (doorState.current_mode.value) {
+    case RomiCore.DoorMode.MODE_OPEN:
+      return classes.open;
+    case RomiCore.DoorMode.MODE_MOVING:
+      return classes.moving;
+    case RomiCore.DoorMode.MODE_CLOSED:
+      return classes.close;
+    default:
+      return classes.unknown;
+  }
+}
+
+interface BaseDoorProps {
+  v1: [number, number];
+  v2: [number, number];
+  className?: string;
+}
+
+const BaseDoor = (props: BaseDoorProps) => {
+  const { v1, v2, className } = props;
+  const classes = useDoorStyles();
+
+  return (
+    <g>
+      <line
+        className={joinClasses(classes.base, className)}
+        x1={v1[0]}
+        y1={v1[1]}
+        x2={v2[0]}
+        y2={v2[1]}
+      />
+    </g>
+  );
+};
+
+interface DummyDoorProps {
+  v1: [number, number];
+  v2: [number, number];
+}
+
+/**
+ * Because we are using stroke-dash in some of the classes, it makes it such that only
+ * the rendered line will be considered for click detection. To workaround it, we use
+ * a transparent door on top of the marker, this dummy door will be used to allow the
+ * full door to be clickable.
+ */
+const DummyDoor = (props: DummyDoorProps) => {
+  const { v1, v2 } = props;
+  const classes = useDoorStyles();
+
+  return (
+    <g>
+      <line
+        className={joinClasses(classes.base, classes.transparent)}
+        x1={v1[0]}
+        y1={v1[1]}
+        x2={v2[0]}
+        y2={v2[1]}
+      />
+    </g>
+  );
+};
+
+type DoorMarkerImplProps = Omit<DoorMarkerProps, 'onClick'>;
+
+/*
+ * Single swing doors:
+ *  - hinge is located at (v1_x, v1_y)
+ *  - door extends till (v2_x, v2_y)
+ *  - motion_range = door swing range in DEGREES
+ *  - there are two possible motions: clockwise and anti-clockwise
+ *  - selected by the motion_direction parameter, which is +1 or -1
+ */
+const SingleSwingDoor = (props: DoorMarkerImplProps) => {
+  const { door, doorState } = props;
+  const doorStyle = useDoorStyle(doorState);
+
+  return (
+    <>
+      <BaseDoor v1={[door.v1_x, door.v1_y]} v2={[door.v2_x, door.v2_y]} className={doorStyle} />
+      <DummyDoor v1={[door.v1_x, door.v1_y]} v2={[door.v2_x, door.v2_y]} />
+    </>
+  );
+};
+
+/*
+ * Single sliding doors:
+ *  - the door slides from (v2_x, v2_y) towards (v1_x, v1_y)
+ *  - range of motion is entire distance from v2->v1. No need to specify.
+ */
+const SingleSlidingDoor = SingleSwingDoor;
+
+/*
+ * single/double telescoping doors:
+ *   * common in elevators; same parameters as sliding doors; they just
+ *     open/close faster and take up less space inside the wall.
+ */
+const SingleTelescopeDoor = SingleSwingDoor;
```

Comment:
I think we are pointing to the same reference here

Reply:
Good catch, it should be referencing to sliding doors.